### ISSUE TITLE
Bugs I have found from Integration 0.77

### ISSUE THREAD

**Issue by eldexterr** [RetroAchievements/RAIntegration#494]
So far I have just found 2, but is always a good idea to document everything, all of this was on RAP64 , I'm using integration 0.77.1
1. If you close the emulator with an address frozed it appear as "Frozen" the new time you load the game but looks like isn't actually frozen, you need to toggle the address to frozen again to make it work

2. When you have an address on the bookmarker you can't edit the value via the memory inspector anymore

**Comment by Keltron3030** [comment]
Similar kind of bug so I'll just add here:

Regarding the memory bookmarker move up/down buttons.  It seems that the behavior column is not behaving as expected.  For example if I have two addresses top frozen and bottom not, if I move the top frozen row down or the bottom non-frozen row up both rows will now show as frozen; however the previously blank row will now show frozen though it's not actually frozen or highlighted as frozen.  If I try to make it blank again there is no change, I have to either freeze or pause it first and then move it back to blank.  So it seems the behavior is still "blank" but the text shows otherwise.  Same case for swapping a pause and a blank.    If one row is paused and the other frozen then everything behaves correctly.  

So basically it seems the problem is that blank rows assume the behavior text of the row they are swapped with, though the true behavior is maintained.

**Comment by Jamiras** [comment]
#1 doesn't make any sense. Frozen state is not saved with the bookmarks, so they should always be unfrozen when loading a game, regardless of how the game was closed. Closing the emulator is an extreme form of closing a game, so there shouldn't even be any residual memory unintentionally affecting things. Please clarify

#2 makes sense if the value is frozen. The memory inspector writes back to the emulator, and the bookmark sees that the emulator memory has changed and restores it to the frozen value. You can edit the value directly in the bookmark to change the frozen value. If the bookmark is not frozen, I cannot reproduce

I can reproduce Keltron's issue. I will investigate.

**Comment by eldexterr** [comment]
1. Hmm... yeah that's correct, I though that's was how I got the problem xd
So just forget about this one, I will try to replicate the problem so I can get back with more precise actions to replicate this one

2. Yeah the value was frozen, didn't happen before the new bookmarker, was very useful when you need to change the values very quickly, here you need to double click and press enter on the value each time you want to change it

**Comment by Jamiras** [comment]
The issue with moving items was that the update-entire-row code was not setting empty fields under the assumption that the row was new. When the code was leveraged to update an entire row after items were rearranged, that logic is no longer applicable.

**Comment by eldexterr** [comment]
Looks like I wasn't ever to replicate problem #1 again xd
I dunno what the hell I made to achieve that, but I found something else

If you are moving through the bookmarker using the arrow keys and you pass through freezed addresses, and after that if you pass through the mouse cursor (if the cursor is located at one bookmark), that bookmark will get selected and you can even freeze the address at the same time beside not having the shift key hold

**Comment by Jamiras** [comment]
The latest issue is a result of the multi-edit code. When the user has multiple items selected and clicks to open the in-place editor, the code prevents the item being clicked from being de-selected. Since the mouse is over an item that could open the in-place editor when navigating away from the item, the de-selection is suppressed. I've modified the logic to only suppress the de-select if CTRL is being pressed.

**Comment by eldexterr** [comment]
I just found a new one, when using the bookmarker when you want to freeze multiple addresses and you have the addresses already selected, if you clic the first selected address(even while holding Shift) to open the drop-list "      , Freeze , Pause" it doesn't appear, all the addresses just get deselect except the first one

**Comment by Jamiras** [comment]
That's Windows doing that. When you shift+select a bunch of records, it's always the first selected record through the clicked record. If you shift+select the first record, you're changing the selection to be the first record through itself.

Using your example above, if you hold shift and click on 0x17ca31, then continue to hold shift and click on 0x17cea4, the last four will be highlighted. If you continue to hold shift and click on 0x17ca15, then the highlight will _change_ to be 0x17ca15 through 0x17ca31, as 0x17ca31 is still considered the start of the highlight, and shift+clicking defines the end to be 0x17ca15. The highlighted records below the starting record will no longer be part of the selection and become unhighlighted.
